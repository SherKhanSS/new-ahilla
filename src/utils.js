export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const getCurentFilm = (films, props) => {
  return films.find((movie) => movie.id === Number(props.match.params.id));
};

export const noop = () => {
  // do nothing
};
