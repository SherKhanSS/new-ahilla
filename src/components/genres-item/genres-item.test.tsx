import * as React from "react";
import * as renderer from "react-test-renderer";
import GenresItem from "./genres-item";
import {noop} from "../../utils";

const genre = `Drama`;

it(`Render GenresItem`, () => {
  const tree = renderer
    .create(<GenresItem
      genre={genre}
      isActive={true}
      onGenresItemClick={noop}
    />, {
      createNodeMock: () => {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
