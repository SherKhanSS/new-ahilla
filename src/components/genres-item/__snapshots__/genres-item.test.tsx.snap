// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render GenresItem 1`] = `
<li
  className="catalog__genres-item catalog__genres-item--active"
  onClick={[Function]}
>
  <a
    className="catalog__genres-link"
    href="#"
  >
    Drama
  </a>
</li>
`;
